templates/pkg/resource/sdk.go.tpl

```diff
@@ -273,3 +273,55 @@ func (rm *resourceManager) terminalAWSError(err error) bool {
 	return false
 {{- end }}
 }
+
+{{- if .CRD.GetImmutableFields }}
+// validateImmutableFields validates the immutable fields and set appropriate condition
+func (rm *resourceManager) validateImmutableFields(ko *svcapitypes.{{ .CRD.Names.Camel }},
+	delta *ackcompare.Delta,) {
+	var fields []string;
+
+	for _, diff := range delta.Differences {
+		if diff.Path.Contains("Spec") {
+		{{- range $immutableField := .CRD.GetImmutableFields }}
+			if diff.Path.Contains("{{$immutableField}}") {
+				fields = append(fields,"{{$immutableField}}")
+			}
+		{{- end }}
+		}
+	}
+
+	// Warning condition
+	var warningCondition *ackv1alpha1.Condition = nil
+	for _, condition := range ko.Status.Conditions {
+		if condition.Type == ackv1alpha1.ConditionTypeWarning {
+			warningCondition = condition
+			break
+		}
+	}
+
+	// Remove the warning condition if warning is no longer present
+	if len(fields) == 0 && warningCondition != nil{
+		var newConditions []*ackv1alpha1.Condition
+		for _, condition := range ko.Status.Conditions {
+			if condition.Type != ackv1alpha1.ConditionTypeWarning {
+				newConditions = append(newConditions,condition)
+			}
+		}
+		ko.Status.Conditions = newConditions
+	}
+
+	if len(fields) > 0 {
+		// Add warning condition
```

Comment:
I'd nix the above comment as well, since it's obvious that is what we're doing here.